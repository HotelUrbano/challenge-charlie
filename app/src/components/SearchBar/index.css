.search-bar {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ffffff;
  height: 65px;
  box-shadow: 0px 6px 3px 0px #0000002b; }
  .search-bar__image {
    background-image: url("../../icons/SVG/44.svg");
    background-position: center;
    background-size: cover;
    margin: 0 10px;
    padding: 25px;
    display: block;
    opacity: 0.4; }
  .search-bar__input {
    font-size: 30px;
    color: #a4a4a3;
    width: 100%;
    height: 100%;
    padding-left: 10px; }
