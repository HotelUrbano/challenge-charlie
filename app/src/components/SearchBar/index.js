import React from 'react';
import './index.css';

const SearchBar = props => {
    return (
        <div className="search-bar">
            <span className="search-bar__image" />
            <input
                type="text"
                onChange={props.change}
                value={props.location}
                className="search-bar__input"
            />
        </div>
    );
};

export default SearchBar;
