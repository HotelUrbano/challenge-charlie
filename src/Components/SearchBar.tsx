import React, { useEffect, useState } from "react"

let timeout: any

function SearchBar({ onSearch }: any, ref: React.ForwardedRef<any>) {
    const [inputValue, setInputValue] = useState<string>('')

    useEffect(() => {
        clearTimeout(timeout);
        timeout = setTimeout(() => {
            if (onSearch) {
                onSearch(inputValue)
            }
        }, 250);

        return () => {
            if (timeout) {
                clearTimeout(timeout)
            }
        }
    }, [inputValue, onSearch])

    return (
        <input
            ref={ref}
            value={inputValue}
            type="text"
            onChange={e => {
                setInputValue(e.target.value)
            }}
        />
    )
}

export default React.forwardRef(SearchBar)