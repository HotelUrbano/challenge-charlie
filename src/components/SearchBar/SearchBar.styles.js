import styled from 'styled-components';
import Devices from '../../const/responsiveness';
import Colors from '../../const/colors';

export const SearchBarContainer = styled.section`
  display: flex;
  width: 100%;
  height: 150px;
`;

export const FormInput = styled.form`
  width: 100%;
`;

export const SearchBarInput = styled.input`
  border: none;
  outline: none;
  height: 100%;
  font-size: 26px;
  padding: 0;
  width: 100%;
  font-family: 'Exo 2', sans-serif;
  font-weight: 300;
  color: ${Colors.GREY};

  @media only screen and (max-width: ${Devices.SMALL_SCREEN}) {
  }

  @media only screen and (min-width: ${Devices.LARGE_SCREEN}) {
    font-size: 36px;
    font-weight: 800;
  }
`;

export const CompassImage = styled.img`
  width: 10%;
  @media only screen and (max-width: ${Devices.LARGE_SCREEN}) {
    color: grey;
    margin: 0 30px;
  }

  @media only screen and (min-width: ${Devices.LARGE_SCREEN}) {
    color: grey;
    margin: 0 50px 0 20px;
  }
`;
