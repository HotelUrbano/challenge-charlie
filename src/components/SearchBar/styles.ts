import styled from 'styled-components'

export const Wrapper = styled.form`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  background-color: #42454e;
  padding: 1.6rem;
  border-radius: 1.2rem 1.2rem 0 0;
  width: 100%;
  max-width: 425px;
  z-index: 2;
`

export const Input = styled.input`
  background-color: #fff;
  color: #42454e;
  flex-grow: 1;
  text-align: center;
  height: 3rem;
  padding: 2rem 1.5rem;
  font-size: 1.3rem;
  border: none;
  border-radius: 1.2rem;
  @media (min-width: 425px) {
    font-size: 1.6rem;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
`

export const Submit = styled.input`
  background-color: #1bba9a;
  color: #fff;
  text-align: center;
  font-weight: bold;
  border-radius: 1.2rem;
  font-size: 1.3rem;
  padding: 0.4rem 1.5rem;
  transition: background-color ease 0.25s;
  margin-left: 1rem;
  margin-top: 1rem;
  cursor: pointer;
  &:hover {
    background-color: #189e83;
  }
  @media (min-width: 320px) {
    margin-left: 0;
  }
  @media (min-width: 425px) {
    font-size: 1.6rem;
    margin-top: 0;
    margin-left: 1rem;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
`
