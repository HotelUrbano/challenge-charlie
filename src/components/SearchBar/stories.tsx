import { Story, Meta } from '@storybook/react/types-6-0'
import SearchBar from '.'

export default {
  title: 'SearchBar',
  component: SearchBar,
  args: {
    title: 'Hello Hurb'
  }
} as Meta

export const Default: Story = (args) => <SearchBar {...args} />
