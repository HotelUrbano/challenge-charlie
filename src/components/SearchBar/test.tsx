import { fireEvent, render, screen } from '@testing-library/react'

import SearchBar from '.'

describe('<SearchBar />', () => {
  it('should render the SearchBar', () => {
    const { container } = render(<SearchBar />)
    expect(screen.getByPlaceholderText('buscar')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should get the Weather when press enter', () => {
    const { container } = render(<SearchBar />)
    fireEvent.keyDown(container, { key: 'Enter', code: 'Enter' })
    expect(container.firstChild).toMatchSnapshot()
  })
})
