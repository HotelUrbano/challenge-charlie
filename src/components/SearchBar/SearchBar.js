import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

//styles
import { SearchBarContainer, SearchBarInput, CompassImage, FormInput } from './SearchBar.styles';

export default function SearchBar({ address, city, setCity, searchCity, loading }) {
	useEffect(() => {
		if (address) {
			setCity(`${address.city}, ${address.state}`);
		}
	}, [address, setCity]);

	return (
		<SearchBarContainer>
			<CompassImage src="public/img/44.svg" />
			<FormInput onSubmit={searchCity}>
				<SearchBarInput
					type="text"
					value={loading ? 'Loading...' : city}
					onChange={(e) => setCity(e.target.value)}
					placeholder="Type your location"
				></SearchBarInput>
			</FormInput>
		</SearchBarContainer>
	);
}

SearchBar.propTypes = {
	address: PropTypes.object,
	city: PropTypes.string,
	setCity: PropTypes.func,
	searchCity: PropTypes.func,
	loading: PropTypes.bool
};
