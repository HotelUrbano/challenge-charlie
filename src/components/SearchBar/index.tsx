import * as S from './styles'

const SearchBar = () => <input placeholder="buscar" />

export default SearchBar
