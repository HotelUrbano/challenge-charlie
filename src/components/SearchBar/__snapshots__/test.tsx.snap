// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should get the Weather when press enter 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<main
  class="c0"
>
  <input
    placeholder="buscar"
  />
</main>
`;

exports[`<SearchBar /> should render the SearchBar 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<main
  class="c0"
>
  <input
    placeholder="buscar"
  />
</main>
`;

exports[`<SearchBar /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<main
  class="c0"
>
  <h1>
    Hello Hurb!
  </h1>
</main>
`;
